src/arithmetic/bound_deducer.cc

```diff
@@ -70,6 +70,40 @@ std::vector<const Object*> GetPath(PrimExpr target, PrimExpr expr) {
   return v.path_;
 }
 
+class BoundRemover : public ExprMutator {
+ public:
+  PrimExpr Remove(const PrimExpr& e) {
+    remove_bounded_ = true;
```

Comment:
Can you document its behavior? Is it still necessary, given bound info is no longer wrapped in the assert_